src/main/java/com/terraformersmc/terrestria/feature/tree/foliageplacers/PredictiveSpruceFoliagePlacer.java

```diff
@@ -55,12 +54,10 @@ protected void generateSquare(ModifiableTestableWorld world, Random random, Tree
 					mutable.set(blockPos, offsetX, offsetY, offsetZ);
 					if (TreeFeature.canReplace(world, mutable)) {
 						actualDistance = calculateActualDistance(offsetX, offsetY, offsetZ, giantTrunk);
-						BlockState baseState = config.leavesProvider.getBlockState(random, mutable);
+						BlockState baseState = config.foliageProvider.getBlockState(random, mutable);
+						// TODO: https://github.com/TerraformersMC/Terrestria/pull/256/files/bf6e019da7969f01ed15f736fbeb0f796d701034#r651410439
 
-						world.setBlockState(mutable, withDistance(baseState, actualDistance), 19);
```

Comment:
In ref to the todo on the added lines. Since these are for the large redwood/hemlock trees I want to be sure that it doesn't break anything

Reply:
I teleported to the according biomes via the `/locatebiome command` and manually grew a few saplings, too, and it all works well. So I guess we can leave it as is

Reply:
Sounds good then